### PR TITLE
Fixed Nar-Sie rising in nullspace if the anchor blood stone got destroyed during Act IV

### LINKED ISSUE #21330: Cult Failure State Doesn't Work
@DeityLink 

Destruction of the bloodstones confirmed. The Cult is no longer an immediate threat to Nanotrasen. Lock down of the station has been revoked.
NAR-SIE HAS RISEN

Doesn't actually make narnar.

### PR DESCRIPTION
**Pull request by DeityLink**
Fixed #21330

:cl:
* bugfix: Fixed Nar-Sie rising in nullspace if the anchor blood stone got destroyed during Act IV.

### CODE REVIEW

**Comment by Kurfursten** [comment]
The other odd thing about this case is that no one was dancing at all. Both @Arthurdentist and I watched the fight and no cultist ever danced (or if they did it was very briefly and immediately interrupted by gunfire and explosions, 3 welderbombs went off by the stone)

File: code/datums/gamemode/factions/bloodcult/bloodcult_buildings.dm [hunk 1]
```diff
@@ -1407,8 +1407,9 @@ var/list/bloodstone_list = list()
 	anchor = FALSE
 	for (var/obj/structure/teleportwarp/TW in src.loc)
 		qdel(TW)
-	new /obj/machinery/singularity/narsie/large(src.loc)
-	stat_collection.cult_narsie_summoned = TRUE
+	if (src && loc)
```
Comment (MadmanMartian) [inline code/datums/gamemode/factions/bloodcult/bloodcult_buildings.dm]: Src can never be null, you'd be better off checking if(!gcDestroyed && loc)

This is evidence of a worse problem involving the while loop never properly breaking. Go to 1382 and change it to while(!gcDestroyed && loc && anchor)

Honestly, this is rather gunked and may need some rework

**Review by MadmanMartian — CHANGES_REQUESTED** [review]

**Comment by DeityLink** [comment]
@Kurfursten The anchor's dance is a bit different from the sacrifice one. The "dance" begins as soon as the blood stone becomes an anchor upon Act IV beginning. This bug never happening before means that this is the first time that the crew actually manages to destroy the anchor. Not bad.

**Review by MadmanMartian — APPROVED** [review]

**Review by MadmanMartian — COMMENTED** [review]
Try and rework it so it doesn't require a while loop.

**Comment by DeityLink** [comment]
I can use a different while loop but what's wrong with while loops specifically?

**Comment by MadmanMartian** [comment]
very dependent on break conditions so as to not kill the serb. Better to avoid them in practice.

**Comment by DeityLink** [comment]
I've tried recursively call a proc instead before but BYOND didn't like that. I could have two procs call each other alternatively instead I guess.

**Comment by Exxion** [comment]
That won't work either
This is fine

**Comment by Kurfursten** [comment]
Wait, you're saying the cult automatically wins a set number of ticks after the spread blood objective?

**Comment by DeityLink** [comment]
@Kurfursten No, the dance only actually progresses if there are dancers. If there are no dancers, progresses slowly rolls back every tick.

**Comment by Kurfursten** [comment]
Alright I'm gonna trust you on this one that it was a fluke even though neither of us saw any dancers. You did test this right to make sure the cult failure happens?

**Comment by DeityLink** [comment]
I'll test tonight.

**Comment by DeityLink** [comment]
For the record I wasn't done testing.

**Comment by DeityLink** [comment]
And just so we're clear, it spawning at ALL when it got destroyed was the real issue here, not just the fact that it spawned in nullspace.

**Comment by MadmanMartian** [comment]
yup

**Pull request merged**